tests/integrations/test_torch.py

```diff
@@ -245,34 +250,121 @@ def test_nested_shape():
-        (torch.Tensor([float("nan"), float("inf"), -float("inf")]), True),
+        (torch.tensor([1.0, 2.0, 3.0]), False),
+        (torch.tensor([0.0, 0.0, 0.0]), False),
+        (torch.tensor([1.0]), False),
+        (torch.tensor([]), True),
+        (torch.tensor([1.0, float("nan"), float("nan")]), False),
+        (torch.tensor([1.0, float("inf"), -float("inf")]), False),
+        (torch.tensor([1.0, float("nan"), float("inf")]), False),
+        (torch.tensor([float("nan"), float("nan"), float("nan")]), True),
+        (torch.tensor([float("inf"), float("inf"), -float("inf")]), True),
+        (torch.tensor([float("nan"), float("inf"), -float("inf")]), True),
+        (torch.tensor([1e31, -1e31], dtype=torch.float32), True),
+        (torch.tensor([1e31, -1e31], dtype=torch.float64), False),
+        (torch.tensor([1e301, -1e301], dtype=torch.float64), True),
     ],
 )
-def test_no_finite_values(test_input, expected, wandb_init_run):
+def test_no_valid_values(test_input, expected, wandb_init_run):
     torch_history = wandb.wandb_torch.TorchHistory(wandb.run.history)
 
-    assert torch_history._no_finite_values(test_input) is expected
+    assert torch_history._no_valid_values(test_input) is expected
 
 
 @pytest.mark.parametrize(
     "test_input,expected",
     [
-        (torch.Tensor([0.0, 1.0, 2.0]), torch.Tensor([0.0, 1.0, 2.0])),
-        (torch.Tensor([1.0]), torch.Tensor([1.0])),
-        (torch.Tensor([0.0, float("inf"), -float("inf")]), torch.Tensor([0.0])),
-        (torch.Tensor([0.0, float("nan"), float("inf")]), torch.Tensor([0.0])),
+        (torch.tensor([0.0, 1.0, 2.0]), torch.tensor([0.0, 1.0, 2.0])),
+        (torch.tensor([1.0]), torch.tensor([1.0])),
+        (torch.tensor([0.0, float("inf"), -float("inf")]), torch.tensor([0.0])),
+        (torch.tensor([0.0, float("nan"), float("inf")]), torch.tensor([0.0])),
+        (
+            torch.tensor([0.0, 1e31, -1e31], dtype=torch.float32),
+            torch.tensor([0.0], dtype=torch.float32),
+        ),
+        (
+            torch.tensor([0.0, 1e31, -1e31], dtype=torch.float64),
+            torch.tensor([0.0, 1e31, -1e31], dtype=torch.float64),
+        ),
+        (
+            torch.tensor([0.0, 1e301, -1e301], dtype=torch.float64),
+            torch.tensor([0.0], dtype=torch.float64),
+        ),
     ],
 )
-def test_remove_infs_nans(test_input, expected, wandb_init_run):
+def test_remove_invalid_entries(test_input, expected, wandb_init_run):
     torch_history = wandb.wandb_torch.TorchHistory(wandb.run.history)
 
-    assert torch.equal(torch_history._remove_infs_nans(test_input), expected)
+    assert torch.equal(torch_history._remove_invalid_entries(test_input), expected)
+
+
+def iter_hist_bounds(dtype: torch.dtype) -> List[Tuple[float]]:
+    if dtype == torch.float32:
+        high = wandb.wandb_torch.HIGH32
+        eps = wandb.wandb_torch.EPS32
+        tiny = wandb.wandb_torch.TINY32
+    else:  # dtype == torch.float64
```

Comment:
is this a safe assumption? Types can a lot of different values. 
Should we be more explicit here?

Reply:
I checked through all of the dtypes [here](https://pytorch.org/docs/stable/tensor_attributes.html), and torch.float32 and torch.float64 are the only two supported by torch.histc, so we'd run into different issues if weights of different dtypes reached this part of the code.